DistroLauncher/ExitStatus.cpp

```diff
@@ -0,0 +1,203 @@
+                }
+
+                // Returns true once we don't find the DistributionInfo::Name in process's stdout
+                auto output = runner.getStdOut();
+                if (output.find(DistributionInfo::Name) == std::wstring::npos) {
+                    return true;
+                }
+
+                // We don't need to be hard real time precise.
+                Sleep(997); // NOLINT(readability-magic-numbers): only used here.
+            }
+            return false;
+        }
+
+        VoidResult act(const KeyValuePairs& launcherCmds)
+        {
+            auto actionIt = launcherCmds.find("action");
+            if (actionIt == launcherCmds.end()) {
+                // nothing to do.
+                return VoidResult();
+            }
+
+            // I'm hardcoding that "action" expects strings.
+            auto action = std::any_cast<std::string>(actionIt->second);
+
+            auto f = capabilities.find(action);
+            if (f == capabilities.end()) {
+                return nonstd::make_unexpected(std::domain_error(action));
+            }
+
+            auto res = f->second();
+            if (!res.has_value()) {
+                return nonstd::make_unexpected(res.error());
+            }
+
+            return VoidResult();
+        }
+
+        // Since there is only one kind configuration entry expected at this moment, it would not make a lot of sense to
+        // create enums or tables to work with that.
+        VoidResult config(const KeyValuePairs& launcherCmds)
+        {
+            auto configIt = launcherCmds.find("defaultUid");
+            if (configIt == launcherCmds.end()) {
+                // nothing to do
+                return VoidResult();
+            }
+            auto default_uid = std::any_cast<unsigned long>(configIt->second);
+            // TODO: Replace the default flags on this call by the current ones reading before writting.
+            // The planned wrapper class should resolve this.
+            auto hr = g_wslApi.WslConfigureDistribution(default_uid, WSL_DISTRIBUTION_FLAGS_DEFAULT);
+            if (FAILED(hr)) {
+                return nonstd::make_unexpected(std::runtime_error(
+                  "Could not configure distro to the new default UID: " + std::to_string(default_uid)));
+            }
+
+            return VoidResult();
+        }
+
+        namespace Actions
+        {
+            VoidResult shutdownDistro()
+            {
+                const std::wstring shutdownCmd = L"wsl -t " + DistributionInfo::Name;
+                int cmdResult = _wsystem(shutdownCmd.c_str());
+                if (cmdResult != 0) {
+                    return nonstd::make_unexpected(std::runtime_error("Failed to invoke shutdown command."));
+                }
+
+                // Before relaunching, give WSL some time to make sure distro is stopped.
+                bool stopSuccess = ensureDistroStopped(30); // NOLINT(readability-magic-numbers): only used here.
```

Comment:
would be perhaps nice to define 30 as a constant at the top to make it easier to find/change later if necessary

Reply:
I was fighting against the linter on this. Your question makes the linter the winner. I'll declare a constant at top, but still inside the anonymous namespace.